src/numeric_index.c

```diff
@@ -198,9 +203,11 @@ NRN_AddRv NumericRangeNode_Add(NumericRangeNode *n, t_docId docId, double value)
 
     // split this node but don't delete its range
     double split = NumericRange_Split(n->range, &n->left, &n->right, &rv);
-
+    // in concurrent mode we don't trim range right away 
+    if (NR_MAX_DEPTH_RANGE == 0 && RSGlobalConfig.concurrentMode == 0) {
+      removeRange(n, &rv);
+    }
```

Comment:
How concurrent mode has something to do with it?
Notice that if there is an issue with concurrent mode then we also have issues with cursors.

Reply:
@MeirShpilraien - As discussed, the tree functions are not thread-safe. See failure - [Link].(https://app.circleci.com/pipelines/github/RediSearch/RediSearch/2451/workflows/cb9945c6-e5ba-4342-867d-80e1912da3b0/jobs/17324)

Reply:
Apparently, unrelated to concurrency.
The issue shows also on master but only with ASAN and sanitizer.